### DISCUSSION TITLE
How to understand `Contribution Point` and `Activation Event`?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [microsoft/vscode-discussions#259] [Extension Development QnA] [answered]
In VSCode Extension, there have below two concepts:
```
Contribution Point
Activation Event
```

I don't understand them, please explain the function of them, why need them?

**Comment by Austindgk232** [comment]
Contribution Point
Activation Event

**Comment by Austindgk232** [comment]
#29

**Comment by gjsjohnmurray** [comment, marked as answer]
When you code an extension it is usually because you want to add (contribute) extra features to VS Code. Your extension uses contribution points to do this.

A lot of what a typical extension adds only happens when the extension has been activated by VS Code. You declare which events in VS Code should activate your extension if it hasn't already been activated since VS Code started or reloaded.

**Reply by liaodalin19903** [reply]
When the extension be activated? Is there a place to list all the scenarios of activing a extension？
